firebase-database/src/main/java/com/google/firebase/database/connection/PersistentConnectionImpl.java

```diff
@@ -1049,6 +1162,34 @@ public void onResponse(Map<String, Object> response) {
         });
   }
 
+  private void sendGet(final Long readId) {
+    hardAssert(canSendReads(), "sendGet called when we can't send gets");
+    OutstandingGet get = outstandingGets.get(readId);
+    if (!get.markSent()) {
+      if (logger.logsDebug()) {
+        logger.debug("get" + readId + " already sent or cancelled, ignoring.");
+        return;
+      }
+    }
+    sendAction(
+        get.getAction(),
+        get.getRequest(),
+        new ConnectionRequestCallback() {
+          @Override
+          public void onResponse(Map<String, Object> response) {
+            OutstandingGet currentGet = outstandingGets.get(readId);
+            if (currentGet == get) {
+              outstandingGets.remove(readId);
+              get.getOnComplete().onResponse(response);
+            } else {
+              if (logger.logsDebug())
+                logger.debug(
+                    "Ignoring on complete for get " + readId + " because it was removed already.");
+            }
```

Comment:
```suggestion
            } else if (logger.logsDebug()) {
                logger.debug(
                    "Ignoring on complete for get " + readId + " because it was removed already.");
            }
```